pandas/io/json/_normalize.py

```diff
@@ -389,6 +389,8 @@ def _pull_field(
         try:
             if isinstance(spec, list):
                 for field in spec:
+                    if result is None:
```

Comment:
While this fixes the issue, the typing suggests `result` should be `dict[str, Any]` from `js`. Is the typing incorrect or the issue more upstream?

Reply:
It's the `Any` type on the js dict: once we're one level down we're not longer guaranteed the result is a Dict. It could be (and in this case is) None.

Reply:
Proper json typing requires recursive types, which seem to be, as of very recently, tentatively supported: https://github.com/python/mypy/issues/731#issuecomment-922760064. This is a whole other kettle of fish, though, that will no doubt reveal a plethora of similar soundness issues.

Reply:
Got it. Could you add an informative message in `KeyError` and test it if possible?

Reply:
done